src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs

```diff
@@ -182,12 +182,25 @@ public ITransformer Load(Stream stream, out DataViewSchema inputSchema)
         }
 
         /// <summary>
-        /// Load the model and its input schema from the stream.
+        /// Load the model and its input schema from the file.
+        /// </summary>
+        /// <param name="modelPath">Path to model.</param>
+        /// <param name="inputSchema">Will contain the input schema for the model. If the model was saved using older APIs
+        /// it may not contain an input schema, in this case <paramref name="inputSchema"/> will be null.</param>
+        /// <returns>The loaded model.</returns>
+        public ITransformer Load(string modelPath, out DataViewSchema inputSchema)
+        {
+            using (var stream = File.OpenRead(modelPath))
+                return Load(stream, out inputSchema);
+        }
+
+        /// <summary>
+        /// Load the model from the stream.
         /// </summary>
         /// <param name="stream">A readable, seekable stream to load from.</param>
         /// <returns>A model of type <see cref="CompositeDataLoader{IMultiStreamSource, ITransformer}"/> containing the loader
         /// and the transformer chain.</returns>
-        public IDataLoader<IMultiStreamSource> Load(Stream stream)
+        public IDataLoader<IMultiStreamSource> LoadDataLoader(Stream stream)
```

Comment:
Should we make a similar change for the `Save` method?

```C#
        /// <summary>
        /// Save the model to the file.
        /// </summary>
        /// <param name="model">The trained model to be saved.</param>
        /// <param name="filePath">Path where model should be saved.</param>
        public void Save<TSource>(IDataLoader<TSource> model, string filePath)
        {
            using (var stream = File.Create(filePath))
                Save(model, stream);
        }
```

These are no longer symmetric.

Reply:
Yes but this is Load API PR. I have created another PR to rename this for Save API https://github.com/dotnet/machinelearning/pull/3018

Reply:
@eerhardt @TomFinley feels we shouldn't change for Save because in .Net we have ReadChar, ReadInt, ReadDouble, etc but Write() is just one loaded set of methods. Thoughts?

Reply:
I dont quite follow is there more discussion on github?

---
In reply to: [267100385](https://github.com/dotnet/machinelearning/pull/3008#discussion_r267100385) [](ancestors = 267100385)